cmd/produce.go

```diff
@@ -30,6 +33,15 @@ Press Ctrl+D to exit.`,
 				}
 				defer producer.Close()
 
+				var schemaID uint32 = 0
+				if "" != encode {
```

Comment:
```suggestion
				if encode != "" {
```

Would keep it consistent with the other comparisons